code/modules/hydroponics/seeds.dm

```diff
@@ -47,8 +48,25 @@
 	if(!icon_harvest && plant_type != PLANT_MUSHROOM && yield != -1)
 		icon_harvest = "[species]-harvest"
 
+	if(!nogenes) // not used on Copy()
+		genes += new /datum/plant_gene/core/lifespan(lifespan)
+		genes += new /datum/plant_gene/core/endurance(endurance)
+		genes += new /datum/plant_gene/core/production(production)
+		if(yield != -1)
+			genes += new /datum/plant_gene/core/yield(yield)
+		if(potency != -1)
+			genes += new /datum/plant_gene/core/potency(potency)
+
+		for(var/p in genes)
+			if(ispath(p))

```

Comment:
It's a list of datums when New() finishes populating it with datums. Until then, it's list of types.